src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -1415,6 +1426,52 @@ public static void applyPartitioner(Config conf)
         paritionerName = partitioner.getClass().getCanonicalName();
     }
 
+    private static DiskAccessMode resolveCommitLogWriteDiskAccessMode(DiskAccessMode providedDiskAccessMode)
+    {
+        boolean compressOrEncrypt = getCommitLogCompression() != null || (getEncryptionContext() != null && getEncryptionContext().isEnabled());
+        boolean directIOSupported = false;
+        try
+        {
+            directIOSupported = FileUtils.getBlockSize(new File(getCommitLogLocation())) > 0;
```

Comment:
I'm not sure - why JDK version? 5.0 runs on 11+ and we do that not because of JDK but rather because that some file systems does not have "block size", for example in memory fs

Reply:
Well, it is still valid - we check if it is supported for the file system rather than for JDK